core/incremental/operator.rs

```diff
@@ -581,34 +582,189 @@ impl IncrementalOperator for FilterOperator {
+                            ) {
+                                columns.push(ProjectColumn {
+                                    expr,
+                                    alias: None,
+                                    compiled,
+                                });
+                            }
+                        }
+                    }
+                    _ => {
+                        // For now, skip TableStar and other cases
+                    }
+                }
+            }
+
+            if columns.is_empty() {
+                // If no columns were extracted, default to projecting all input columns
+                input_column_names
+                    .iter()
+                    .filter_map(|name| {
+                        let expr = Expr::Id(Name::Ident(name.clone()));
+                        CompiledExpression::compile(
+                            &expr,
+                            &input_column_names,
+                            schema,
+                            &temp_syms,
+                            internal_conn.clone(),
+                        )
+                        .ok()
+                        .map(|compiled| ProjectColumn {
+                            expr,
+                            alias: None,
+                            compiled,
+                        })
+                    })
+                    .collect()
+            } else {
+                columns
+            }
+        } else {
+            // Not a simple SELECT statement, default to projecting all columns
+            input_column_names
+                .iter()
+                .filter_map(|name| {
+                    let expr = Expr::Id(Name::Ident(name.clone()));
+                    CompiledExpression::compile(
+                        &expr,
+                        &input_column_names,
+                        schema,
+                        &temp_syms,
+                        internal_conn.clone(),
+                    )
+                    .ok()
+                    .map(|compiled| ProjectColumn {
+                        expr,
+                        alias: None,
+                        compiled,
+                    })
+                })
+                .collect()
+        };
+
+        // Generate output column names based on aliases or expressions
         let output_column_names = columns
             .iter()
-            .map(|c| match c {
-                ProjectColumn::Column(name) => name.clone(),
-                ProjectColumn::Expression { alias, .. } => {
-                    alias.clone().unwrap_or_else(|| "expr".to_string())
-                }
+            .map(|c| {
+                c.alias.clone().unwrap_or_else(|| {
+                    // For simple column references, use the column name
+                    if let Expr::Id(name) = &c.expr {
+                        name.as_str().to_string()
+                    } else {
+                        "expr".to_string()
```

Comment:
What about `Qualified`? Also, `Expr` implements `Display` so you could just call `expr.to_string()`

Reply:
And DoublyQualified too